tests/fixtures/site.py

```diff
@@ -1,41 +1,150 @@
+            * len(resources)))
+
+
+def _create_resources(factories, resource_kwargs, translated_resource_kwargs):
     resources = factories['resource'](batch_kwargs=resource_kwargs)
     for i, translated_resource in enumerate(translated_resource_kwargs):
-        translated_resource.update(
-            dict(resource=resources[i % len(resources)]))
-    translated_resources = factories['translated_resource'](
-        batch_kwargs=translated_resource_kwargs)
+        resource = _get_resource_key(resources, translated_resource_kwargs, i)
+        translated_resource.update(dict(resource=resources[resource]))
+    return resources, translated_resource_kwargs
+
+
+def _get_translation(entities, locale, i):
+    # spread the translations over the previous 17 days
+    return dict(
+        date=timezone.make_aware(
+            datetime.now()
+            - timedelta(days=i % 17)),
+        entity=entities[i % len(entities)],
+        locale=locale)
+
+
+def _create_translations(factories, resources, locales):
+    entities = factories['entity'](
+        batch_kwargs=[
+            dict(resource=resources[i % len(resources)])
+            for i in range(0, 2 * len(resources))])
+    translation_batch_kwargs = []
+    translation_count = len(locales) * len(entities) * 2
+    for i in range(0, translation_count):
+        # spread the translations between locales
+        locale = int(
+            math.floor(float(i) / translation_count * len(locales)))
+        translation_batch_kwargs.append(
+            _get_translation(entities, locales[locale], i))
+    translations = factories['translation'](
+        batch_kwargs=translation_batch_kwargs)
+    return entities, translations
+
+
+def _get_latest_translations(translations):
+    latest_translations = {}
+    for translation in translations:
+        resource = translation.entity.resource.pk
+        locale = translation.locale.code
+        latest_translations[resource] = latest_translations.get(resource, {})
+        date, _translation = latest_translations[resource].get(
+            locale,
+            (timezone.make_aware(datetime.min), None))
+        if translation.date > date:
+            latest_translations[resource][locale] = (
+                translation.date, translation)
+    return latest_translations
+
+
+def _get_latest_translation(latest, kwargs):
+    return latest.get(kwargs['resource'].pk, {}).get(
+        kwargs['locale'].code, (None, None))[1]
+
+
+def _create_translated_resources(factories, translations,
+                                 translated_resource_kwargs):
+    latest = _get_latest_translations(translations)
+    for kwargs in translated_resource_kwargs:
+        kwargs["latest_translation"] = _get_latest_translation(latest, kwargs)
+    translated_resources = (
+        factories['translated_resource'](
+            batch_kwargs=translated_resource_kwargs))
+    return translated_resources
+
+
+@pytest.fixture
+def site_matrix(factories):
```

Comment:
This file could do with a lot more docstrings on functions, especially this one and `_create_site_root`. I'm not exactly sure what you mean by "site matrix" and "site root", so a short explanation could be quite useful. :)

Reply:
added, hope its enough to make this code a bit more readable